### DISCUSSION TITLE
Pole techniczne w ofercie do przechowywania własnego identyfikatora (dublowanie ofert)

### DISCUSSION THREAD

**Discussion by Onixarts** [allegro/allegro-api#8259] [Suggestions - rejected]
**HISTORIA**
Kiedyś dawno w starym API Allegro było pole o nazwie localId bodajże, w którym mogliśmy umieszczać jakiś id z własnego systemu sprzedawcy. Używaliśmy tego pola do eliminowania dubli ofert przy wystawianiu – które mogły się pojawiać przy niestabilnym połączeniu, timeoutach itp.

Potem nastało nowe API i tam problem z dublami trochę zniknął dzięki temu, że były idempotentne komendy aktywacji, które eliminowały problem.

Teraz, gdy pojawiło się jeszcze nowsze API i zasoby sale/product-offers jako takiej komendy aktywacji nie ma, bo oferta się sama aktywuje.

**PROBLEM**
No i tutaj rodzi się problem z dublami, ponieważ nie mamy jak przekazać jakiegoś identyfikatora, który pozwoliłby nam ustalić, że oferta została już wystawiona, pomimo, że z takich czy innych przyczyn response do nas nie wrócił, lub jego zapis się nie powiódł w systemie użytkownika.

Jest co prawda pole z Sygnaturą, ale pole to jest do dyspozycji sprzedawcy i wrzucanie tam jakichś naszych, technicznych identyfikatorów przeszkadza w użytkowaniu tego pola (np. przy wyszukiwaniu czegoś po stronie serwisu Allegro po tym polu).

**SUGESTIA**
Prośba o dodanie takiego technicznego pola local-id do oferty, po którym to polu była by możliwość wyszukiwania tak aby w razie niejasnej sytuacji, czy oferta się wystawiła, móc strzelić do serwisu z zapytaniem czy jest taka oferta wystawiona. 

Na nowy zasobie mamy już zgłoszenia od użytkowników, że takie duble w ofertach po stronie Allegro potrafią się pojawić, no bo nie bardzo mamy jak zweryfikować czy oferta się wystawiła bez śmiecenia użytkownikom w danych widocznych na stronie oferty albo w panelu.

**Comment by jezewskiBNetrox** [comment]
> Teraz, gdy pojawiło się jeszcze nowsze API i zasoby sale/product-offers jako takiej komendy aktywacji nie ma, bo oferta się sama aktywuje.

Zastanawiam się czy tego zasobu nie dałoby się używać tak jak /sale/offers. Nie jestem pewien czy przeszłoby tworzenie oferty bez planowania publikacji ale utworzenie szkicu powinno być jak najbardziej możliwe. Potem używając https://developer.allegro.pl/documentation#operation/changePublicationStatusUsingPUT można by opublikować szkic (tj. zaplanować czy bądź aktywować go).

Nie wiem jak na taki sposób użycia zapatrują się deweloperzy, jednak jeżeli nie usuwali tego zasobu to pozostawienie funkcjonalności szkiców i ich aktywacji powinno pozostać.

**Comment by PrzemyslawLukanowski** [comment]
Dokładnie tak jak napisał @jezewskiBNetrox - możecie [utworzyć najpierw szkic oferty](https://developer.allegro.pl/tutorials/jak-jednym-requestem-wystawic-oferte-powiazana-z-produktem-D7Kj9gw4xFA#szkic-oferty), nie aktywować jej w ramach tego samego requestu, ale dopiero w następnym krok, według poprzedniego flow, czyli za właśnie pomocą https://developer.allegro.pl/documentation#operation/changePublicationStatusUsingPUT. Czy to rozwiązałoby Wasz aktualny problem?

**Comment by Onixarts** [comment]
Teoretycznie tak, aczkolwiek, nie rozwiązuje to problemu tworzenia nadmiarowych szkiców :), których liczba ograniczona jest chyba do 20k - co już udało się niektórym użytkownikom w przeszłości osiągnąć.

**Comment by PrzemyslawLukanowski** [comment]
Z uwagi na istniejące obejście problemu, na tę chwilę nie planujemy wdrażać takiego dodatkowego pola.
